packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol

```diff
@@ -38,9 +39,9 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
      * Constants *
      *************/
 
-    // TODO: should be the amount sufficient to cover the gas costs of all of the transactions up
+    // A value that is large enought to account for the gas usage of the transaction
     // to and including the CALL/CREATE which forms the entrypoint of the transaction.
-    uint256 constant EXECUTION_VALIDATION_GAS_OVERHEAD = 25000;
+    uint256 constant EXECUTION_VALIDATION_GAS_OVERHEAD = 55000;
```

Comment:
Hmm, I think that what this original comment with "up to and including" meant was that it includes the gas cost consumed directly by `EM.ovmCALL/ovmCREATE`, but NOT including the gas consumed by the subcall that is made by the EM out to the `tx.to` OVM contract.  This is why we subtract out the `gasLimit` in the check below, because that subcall to `tx.to` consumes at most `gasLimit` additional gas.  I feel like we could say
```
to and INCLUDING the containerization cost of CALL/CREATE, but EXCLUDING the subcall made by the Execution Manager, which will consume at most gasLimit additional gas.
```

Reply:
Also, [this comment](https://github.com/ethereum-optimism/optimism/pull/1008#discussion_r647940058) still applies.  Because we subtract out the `gasleft` only [after](https://github.com/ethereum-optimism/optimism/blob/5ec3244b7a4b8da709095f7a745e9781c9e9e8b1/packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol#L79) we do the `tx.decode`.  This means that the value as currently used is really just the gas for [this subset of `execute`](https://github.com/ethereum-optimism/optimism/blob/5ec3244b7a4b8da709095f7a745e9781c9e9e8b1/packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol#L83-L157), not the full thing.  The approach I outlined in the linked github comment would make this comment more accurate, although the two are equivalent aside from a constant.

Reply:
> Oh, good point. Maybe the correct way to do it might be have uint initialGas = gasleft() at the very first line of execute, and then here check that initialGas - intrinsicGas >= gasLimit? That way the decoding and intrinsic gas calculation can be considered "a part of" the intrinsic gas.

I'm not sure if I follow - what exactly is the point of the intrinsic gas? I feel like we just need to require that there is enough gas left so that the code makes it to the subcall